src/use-map.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useMemo } from 'react';
+
+export const useMap = <TKey, TValue>(initialState = new Map<TKey, TValue>()) => {
+  const [value, setValue] = useState(initialState);
+
+  const handlers = useMemo(() => Object.assign(value, {
+    set: (key: TKey, value: TValue) => {
+      setValue((oldMap: Map<TKey, TValue>): Map<TKey, TValue> => {
```

Comment:
I think you can omit the return type annotation here since the TS compiler can infer it

Reply:
yes, no probs